Code/Framework/AzCore/Tests/AZStd/CreateDestroy.cpp

```diff
@@ -134,4 +134,46 @@ namespace UnitTest
         EXPECT_FLOAT_EQ(4.0f, resultAddress->m_floatValue);
         AZStd::destroy_at(resultAddress);
     }
+
+    TEST(CreateDestroy, UninitializedDefaultConstruct_IsAbleToConstructMultipleElements_Succeeds)
+    {
+        struct RefWrapper
+        {
+            int m_intRef{ 2 };
+        };
+        constexpr size_t ArraySize = 2;
+        AZStd::aligned_storage_for_t<RefWrapper> testArray[ArraySize];
+        RefWrapper(&uninitializedAddress)[2] = reinterpret_cast<RefWrapper(&)[2]>(testArray);
+        AZStd::uninitialized_default_construct(AZStd::begin(uninitializedAddress), AZStd::end(uninitializedAddress));
+
+
+        EXPECT_EQ(2, uninitializedAddress[0].m_intRef);
+        EXPECT_EQ(2, uninitializedAddress[1].m_intRef);
+        // Reset uninitializedAddress to Debug pattern
+        memset(uninitializedAddress, 0xCD, ArraySize * sizeof(RefWrapper));
+        AZStd::uninitialized_default_construct_n(AZStd::data(uninitializedAddress), AZStd::size(uninitializedAddress));
+        EXPECT_EQ(2, uninitializedAddress[0].m_intRef);
+        EXPECT_EQ(2, uninitializedAddress[1].m_intRef);
+    }
+
+    TEST(CreateDestroy, UninitializedValueConstruct_IsAbleToConstructMultipleElements_Succeeds)
+    {
+        struct RefWrapper
+        {
+            int m_intRef{ 2 };
```

Comment:
It might be useful to add a non-trivial constructor here which assigns `m_intRef` to something to disambiguate this test case (testing value construction) from the default construction case

Reply:
Yes I will go ahead and do that.

Reply:
Actually there is a bit of an problem testing the default construction case above.

Adding a non-trivial constructor here without a default member initializer would make `m_intValue` initialize to an indeterminate value.

So going with something such as 
```
struct RefWrapper
{
    RefWrapper()
    {}
    int m_intValue;
};
```
Results in `m_intValue` being indeterminate

Because it isn't possible to "test" if a value is indeterminate, they only thing that can be tested is an actual value.

Using member initialization with
```
struct RefWrapper
{
    RefWrapper()
        : m_intValue{}
    {}
    int m_intValue;
};
```
Will use value initialization for `m_intValue` which results in it being zero initialized, but in that case default initialization would also do the same.

It is actually not possible to contrive a case where default initialization constructs a member to a known value and value initialization constructs a member to a different known value.

Using a structure such as the follows is better for value-initialization, as it would result in zero-initialization occuring for the RefWrapper class.
```
struct RefWrapper
{
    int m_intValue;
};
```

If default initialization was used on the above struct, it would result in an indeterminate value for `m_intValue`, but that "indeterminism" can't be tested